src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.SetTimes.OSX.cs

```diff
@@ -2,12 +2,19 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
 
 namespace System.IO
 {
     internal partial struct FileStatus
     {
         internal void SetCreationTime(string path, DateTimeOffset time, bool asDirectory)
+            => SetCreationTime(handle: null, path, time, asDirectory);
+
+        internal void SetCreationTime(SafeFileHandle handle, DateTimeOffset time, bool asDirectory)
+            => SetCreationTime(handle, handle.Path, time, asDirectory);
```

Comment:
I had not compiled it on OSX. `handle.Path` needs to be replaced by `GetHandlePath(handle)` here.